src/Draco.LanguageServer/Handlers/DracoHoverHandler.cs

```diff
@@ -0,0 +1,62 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Draco.Compiler.Api;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Draco.LanguageServer.Handlers;
+
+internal class DracoHoverHandler : HoverHandlerBase
+{
+    private readonly DracoDocumentRepository documentRepository;
+
+    public DracoHoverHandler(DracoDocumentRepository documentRepository)
+    {
+        this.documentRepository = documentRepository;
+    }
+
+    protected override HoverRegistrationOptions CreateRegistrationOptions(HoverCapability capability, ClientCapabilities clientCapabilities) => new()
+    {
+        DocumentSelector = new DocumentSelector(new DocumentFilter
+        {
+            Pattern = $"**/*{Constants.DracoSourceExtension}",
+        })
```

Comment:
We got a PR merged that factors out the document selectors, check out the `Constants` class